internal/storage/clickhouse.go

```diff
@@ -756,3 +757,77 @@ func (c *ClickHouseConnector) GetTraces(qf QueryFilter) (traces []common.Trace,
 	}
 	return traces, nil
 }
+
+// TODO make this atomic
+func (c *ClickHouseConnector) InsertDataForBlocks(data []common.BlockData) error {
+	blocks := make([]common.Block, 0, len(data))
+	logs := make([]common.Log, 0)
+	transactions := make([]common.Transaction, 0)
+	traces := make([]common.Trace, 0)
+
+	for _, blockData := range data {
+		blocks = append(blocks, blockData.Block)
+		logs = append(logs, blockData.Logs...)
+		transactions = append(transactions, blockData.Transactions...)
+		traces = append(traces, blockData.Traces...)
```

Comment:
should we delete blockData after finishing here, as it's not needed anywhere and only blows the service memory

Reply:
it is used for metrics. I think it's dangerous to delete data manually inside the storage layer if it doesn't have context about business logic